haystack/nodes/prompt/providers.py

```diff
@@ -261,22 +257,14 @@ def _ensure_token_limit(self, prompt: str) -> str:
 
     @classmethod
     def supports(cls, model_name_or_path: str, **kwargs) -> bool:
+        task_name: Optional[str] = None
         try:
-            config = AutoConfig.from_pretrained(model_name_or_path)
-        except OSError:
-            # This is needed so OpenAI models are skipped over
+            task_name = get_task(model_name_or_path)
```

Comment:
Nice! I like this more than the AutoConfig loading. Does `get_task` require downloading a file in the background? 

Also does `get_task` support `use_auth_token`? If so, it might be worth adding here so private models also work with PromptNode even though that is probably an unlikely case at the moment.

Reply:
Yeah it's RPC call to hugging face hub. Perhaps it caches some data, and if it doesn't we can map some well-known models. Others can go on the HF hub. Good points - will check.

Reply:
HF ppl were quick to fix mt0 mislabeling as well @sjrl See https://huggingface.co/bigscience/mt0-small/discussions/2#640890c904cad621ef9120b3